### PR TITLE
cmake: boilerplate: improve readability

### PR DESCRIPTION
**Pull request by mped-oticon**
Following the same convention as that of ccache, finding python and git are moved out to separate cmake files.

The main point of this is to improve readability of `boilerplate.cmake` file and separate concerns.

### CODE REVIEW

File: cmake/git.cmake [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Use git if it is installed, to set BUILD_VERSION
+
+find_package(Git QUIET)
```
Comment (SebastianBoe) [inline cmake/git.cmake]: I can't see the patch that removes 'find_package(Git' from where this came from.

**Review by SebastianBoe — CHANGES_REQUESTED** [review]

File: cmake/git.cmake (hunk 1 above)
Comment (mped-oticon) [reply cmake/git.cmake]: Fixed.

File: cmake/git.cmake [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  execute_process(COMMAND ${GIT_EXECUTABLE} describe
+    WORKING_DIRECTORY                ${ZEPHYR_BASE}
+    OUTPUT_VARIABLE                  BUILD_VERSION
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+    ERROR_STRIP_TRAILING_WHITESPACE
+    ERROR_VARIABLE                   stderr
+    RESULT_VARIABLE                  return_code
+  )
+  if(return_code)
+    message(STATUS "git describe failed: ${stderr}; ${KERNEL_VERSION_STRING} will be used instead")
```
Comment (SebastianBoe) [inline cmake/git.cmake]: 'KERNEL_VERSION_STRING'

is set by version.cmake. This patch moves git.cmake before version.cmake, making this variable unavailable.

Could you also check other variables? (or retain the location)

**Review by SebastianBoe — CHANGES_REQUESTED** [review]

File: cmake/git.cmake (hunk 2 above)
Comment (mped-oticon) [reply cmake/git.cmake]: Correct. I looked for other use-before-def, but could only find that.
I've moved inclusion of version.cmake up to be earlier.

File: cmake/app/boilerplate.cmake [hunk 3]
```diff
@@ -88,22 +88,21 @@ set(AUTOCONF_H ${__build_dir}/include/generated/autoconf.h)
...
-set(PythonInterp_FIND_REQUIRED 1)
-include(${ZEPHYR_BASE}/cmake/backports/FindPythonInterp.cmake)
+#
+# Import more CMake functions and macros
+#
 
 include(CheckCCompilerFlag)
 include(CheckCXXCompilerFlag)
 include(${ZEPHYR_BASE}/cmake/extensions.cmake)
+include(${ZEPHYR_BASE}/cmake/version.cmake)  # depends on hex.cmake
```
Comment (SebastianBoe) [inline cmake/app/boilerplate.cmake]: We are saying 'depends on x' here and below.

But in one case this means that we include x, and in the other that we required x to already be included.

We need to use different language for these two different situations.

**Review by SebastianBoe — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**